cupy/_core/_carray.pyx

```diff
@@ -10,8 +10,12 @@ cdef class CArray(function.CPointer):
 
     cdef void init(
             self, void* data_ptr, Py_ssize_t data_size,
-            const shape_t& shape, const strides_t& strides):
+            const shape_t& shape, const strides_t& strides) except*:
         cdef size_t ndim = shape.size()
+        assert ndim == strides.size()
+        if ndim > MAX_NDIM:
```

Comment:
I feel `assert ndim <= MAX_NDIM` is enough as these classes are private and not constructed by users.

Reply:
Good point. I don't remember what I was thinking, but indeed `CArray` and `CIndexer` are not constructed by users.

Reply:
Fixed in f534a49.